eng/pipelines/vmr-build-pr.yml

```diff
@@ -66,6 +66,12 @@ stages:
       ${{ else }}:
         scope: lite
 
+  - ${{ if eq(variables['isSourceOnlyBuild'], false) }}:
+    - template: templates/stages/vmr-final-join.yml
+      parameters:
+        azureDevOpsOrg: dnceng-public
+        azureDevOpsProject: public
```

Comment:
Why do we import this here instead of stages/vmr-build.yml? That file lists all the jobs including the join jobs that will eventually add and is the source of truth for the list of verticals.

Reply:
I see this `Final Join` stage as a `Post build` stage, where we prepare for BAR publishing and eventually (as a part of the next PR) do the BAR publishing, so not putting in vmr-build made sense to me

Reply:
I did some refactoring, renamed the stage to `VMR Post Build` for better clarity. I think with that name it becomes clear that it should be a separate yml. Let me know if you think differently please